log4j2-propagation/src/main/java/nl/talsmasoftware/context/log4j2/threadcontext/Log4j2ThreadContextManager.java

```diff
@@ -56,42 +54,52 @@
  * prefer using this library to cover all use cases.
  * <p>
  * As with all manager implementations of this library there is usually no need to directly
- * interact with the manager classes. Instead Java's {@link ServiceLoader} makes sure they
+ * interact with the manager classes. Instead Java's {@code ServiceLoader} makes sure they
  * are loaded as services. If an instance of this class is needed nonetheless it can be obtained
- * through the field {@link #INSTANCE}.
+ * through the {@link #provider()} method.
  *
  * @see <a href="https://logging.apache.org/log4j/2.x/manual/thread-context.html">Log4j 2 Thread Context manual</a>
  */
-public class Log4j2ThreadContextManager implements ClearableContextManager<Log4j2ThreadContextData> {
+public class Log4j2ThreadContextManager implements ClearableContextManager<Log4j2ThreadContextSnapshot> {
     /**
      * Singleton instance of this class.
      */
-    public static final Log4j2ThreadContextManager INSTANCE = new Log4j2ThreadContextManager();
+    private static final Log4j2ThreadContextManager INSTANCE = new Log4j2ThreadContextManager();
 
     /**
-     * Returns the singleton instance.
+     * Returns the singleton instance of the {@linkplain Log4j2ThreadContextManager}.
      * <p>
-     * This method mainly exists for usage by {@link ServiceLoader}. The singleton instance
-     * can also directly be obtained from {@link #INSTANCE}.
+     * The ServiceLoader supports a static {@code provider()} method to resolve services since Java 9.
```

Comment:
Good point, but even the `provider()` method won't be used by most users, as they can simply use the static `ContextManagers` methods.
I tried your suggestion but really dislike the javadoc + inline comment combination so I'll leave the message in the regular documentation, I doubt it will bother many 😉